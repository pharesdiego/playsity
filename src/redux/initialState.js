const initialState = {
  videoplayer: {
    url: '/video.mp4'
  },
  playlist: {
    clips: [
      {
        id: null,
        name: 'Full Video',
        start: '0',
        end: '0',
        tags: ''
      }
    ],
    currentClip: null
  }
};

export default initialState;