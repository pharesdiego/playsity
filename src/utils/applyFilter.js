const applyFilter = (clips = [], filter = '') => {
  if(!filter) return clips;
  filter = filter.toLowerCase().trim();
  return clips
          .filter(clip =>
                    clip.name.toLowerCase().includes(filter) || 
                    clip.tags.split(',').some(tag => tag.toLowerCase().includes(filter))
                  );
}

export default applyFilter;