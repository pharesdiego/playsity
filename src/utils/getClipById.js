const getClipById = (clips = [], id = false) => {
  if(!id) return {};
  let clip = clips.find(clip => clip.id === id);
  return clip || {};
}

export default getClipById;