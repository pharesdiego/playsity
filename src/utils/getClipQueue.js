const getClipQueue = (clips = [], currentClip) => {
  let index = clips.findIndex(clip => clip.id === currentClip);
  if(index === -1) return {};
  return {
    previous: index === 0 ? null : clips[index - 1],
    next: index === clips.length - 1 ? null : clips[index + 1]
  }
}

export default getClipQueue;