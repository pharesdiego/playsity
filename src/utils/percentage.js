const percentage = (time, start, end) => {
  if(time < start) return 0;
  if(time > end) return 100;
  return ((time - start) * 100) / (end - start);
};

export default percentage;