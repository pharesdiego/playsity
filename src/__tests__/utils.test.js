import applyFilter from './../utils/applyFilter';
import getClipById from './../utils/getClipById';
import getClipQueue from './../utils/getClipQueue';
import percentage from './../utils/percentage';
import initialState from './../redux/initialState';

const clips = initialState.playlist.clips;

describe('>>> Utils Tests <<<', () => {
  it('applyFilter', () => {
    expect(applyFilter()).toEqual([]);
    expect(applyFilter([])).toEqual([]);
    expect(applyFilter(clips, '|@!&^.,Ã±')).toEqual([]);
    expect(applyFilter(clips)).toMatchSnapshot();
    expect(applyFilter(clips, 'intro')).toMatchSnapshot();
    expect(applyFilter(clips, 'quest')).toMatchSnapshot();
  });

  it('getClipById', () => {
    expect(getClipById()).toEqual({});
    expect(getClipById(clips)).toEqual({});
    expect(getClipById(clips, '?.@!!@#$%!@#:')).toEqual({});
    expect(getClipById(clips, 'a21e19da2dc')).toMatchSnapshot();
    expect(getClipById(clips, '9ea3abcc842')).toMatchSnapshot();
  });

  it('getClipQueue', () => {
    expect(getClipQueue()).toEqual({});
    expect(getClipQueue(clips)).toEqual({});
    expect(getClipQueue(clips, null)).toMatchSnapshot();
    expect(getClipQueue(clips, 'a21e19da2dc')).toMatchSnapshot();
    expect(getClipQueue(clips, '9ea3abcc842')).toMatchSnapshot();
  });

  it('percentage', () => {
    expect(percentage(10, 10, 30)).toBe(0);
    expect(percentage(-10, 10, 30)).toBe(0);
    expect(percentage(20, 0, 40)).toBe(50);
    expect(percentage(10, 0, 100)).toBe(10);
    expect(percentage(30, 0, 100)).toBe(30);
    expect(percentage(44, 0, 100)).toBe(44);
    expect(percentage(60, 0, 100)).toBe(60);
    expect(percentage(70, 0, 100)).toBe(70);
    expect(percentage(700, 0, 100)).toBe(100);
  });
});