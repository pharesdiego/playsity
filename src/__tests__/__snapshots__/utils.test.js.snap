// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Utils Tests <<< applyFilter 1`] = `
Array [
  Object {
    "end": "52",
    "id": null,
    "name": "Full Video",
    "start": "0",
    "tags": "",
  },
  Object {
    "end": "11",
    "id": "a21e19da2dc",
    "name": "Intro",
    "start": "0",
    "tags": "Snow, Walking, Mountains",
  },
  Object {
    "end": "45",
    "id": "9ea3abcc842",
    "name": "A dangerous quest",
    "start": "35",
    "tags": "",
  },
]
`;

exports[`>>> Utils Tests <<< applyFilter 2`] = `
Array [
  Object {
    "end": "11",
    "id": "a21e19da2dc",
    "name": "Intro",
    "start": "0",
    "tags": "Snow, Walking, Mountains",
  },
]
`;

exports[`>>> Utils Tests <<< applyFilter 3`] = `
Array [
  Object {
    "end": "45",
    "id": "9ea3abcc842",
    "name": "A dangerous quest",
    "start": "35",
    "tags": "",
  },
]
`;

exports[`>>> Utils Tests <<< getClipById 1`] = `
Object {
  "end": "11",
  "id": "a21e19da2dc",
  "name": "Intro",
  "start": "0",
  "tags": "Snow, Walking, Mountains",
}
`;

exports[`>>> Utils Tests <<< getClipById 2`] = `
Object {
  "end": "45",
  "id": "9ea3abcc842",
  "name": "A dangerous quest",
  "start": "35",
  "tags": "",
}
`;

exports[`>>> Utils Tests <<< getClipQueue 1`] = `
Object {
  "next": Object {
    "end": "11",
    "id": "a21e19da2dc",
    "name": "Intro",
    "start": "0",
    "tags": "Snow, Walking, Mountains",
  },
  "previous": null,
}
`;

exports[`>>> Utils Tests <<< getClipQueue 2`] = `
Object {
  "next": Object {
    "end": "45",
    "id": "9ea3abcc842",
    "name": "A dangerous quest",
    "start": "35",
    "tags": "",
  },
  "previous": Object {
    "end": "52",
    "id": null,
    "name": "Full Video",
    "start": "0",
    "tags": "",
  },
}
`;

exports[`>>> Utils Tests <<< getClipQueue 3`] = `
Object {
  "next": null,
  "previous": Object {
    "end": "11",
    "id": "a21e19da2dc",
    "name": "Intro",
    "start": "0",
    "tags": "Snow, Walking, Mountains",
  },
}
`;
